piqueserver/game_modes/tdm.py

```diff
@@ -65,30 +87,53 @@ def get_kill_count(self):
                 return ("Green leads %s-%s (+%s, %s left). Playing to %s kills." %
                         (green_kills, blue_kills,
                          diff,
-                         self.kill_limit - green_kills,
-                         self.kill_limit))
+                         KILL_LIMIT.get() - green_kills,
+                         KILL_LIMIT.get()))
             elif green_kills < blue_kills:
                 return ("Blue leads %s-%s (+%s, %s left). Playing to %s kills." %
                         (blue_kills, green_kills,
                          -diff,
-                         self.kill_limit - blue_kills,
-                         self.kill_limit))
+                         KILL_LIMIT.get() - blue_kills,
+                         KILL_LIMIT.get()))
             else:
                 return ("%s-%s, %s left. Playing to %s kills." %
                         (green_kills,
                          blue_kills,
-                         self.kill_limit - green_kills,
-                         self.kill_limit))
+                         KILL_LIMIT.get() - green_kills,
+                         KILL_LIMIT.get()))
+
+        # since its a team based game, we gonna split the caps
+        # for all players in the team
+        def do_captures(self, team, caps):
+            while (team.score < caps):
+                for player in team.get_players():
+                    if team.score >= caps:
+                        break
+
+                    team.score += 1
+                    intel_capture = IntelCapture()
+                    intel_capture.player_id = player.player_id
+                    intel_capture.winning = False
+
+                    self.broadcast_contained(intel_capture)
 
         def check_end_game(self, player):
-            if self.green_team.kills >= self.kill_limit:
-                self.send_chat("Green Team Wins, %s - %s" %
-                               (self.green_team.kills, self.blue_team.kills))
+            if SCORE_PERCENTAGE.get() and player:
+                team = player.team
+                caps_percent = math.floor(
+                    self.max_score*team.kills/KILL_LIMIT.get())
+
+                if caps_percent > team.score:
+                    self.do_captures(team, caps_percent)
+
+            if self.green_team.kills >= KILL_LIMIT.get():
+                self.broadcast_chat("Green Team Wins, %s - %s" %
```

Comment:
It might actually be neat to switch this to use the actual team names in the future